### PR TITLE
template(remix): Add @typescript-eslint/parser to devDependencies

### PR DESCRIPTION
**Pull request by hpnzm**
Eslint doesn't work without ```@typescript-eslint/parser``` and throw error when run ```yarn lint```. So this PR add the missing ```@typescript-eslint/parser``` dependency to the template.

### CODE REVIEW

File: templates/remix/package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "@types/react": "^18.2.20",
     "@types/react-dom": "^18.2.7",
     "@typescript-eslint/eslint-plugin": "^6.7.4",
+    "@typescript-eslint/parser": "^7.0.1",
```
Comment (brophdawg11) [inline templates/remix/package.json:26]: I think we should sync this with the semver of the plugin itself - the carat should ensure it gets the latest within the `6.x` semver release

```suggestion
    "@typescript-eslint/parser": "^6.7.4",
```

File: contributors.yml [hunk 2]
```diff
@@ -646,4 +647,4 @@
 - zachdtaylor
 - zainfathoni
 - zayenz
-- zhe
+- zhe
```
Comment (brophdawg11) [inline contributors.yml:650]:
```suggestion
- zhe

```

Comment (brophdawg11) [reply contributors.yml:650]: We can keep the trailing newline here

**Comment by brophdawg11** [comment]
oh interesting - does `yarn` not install `peerDependencies` by default?  This has come up before but was not an issue with `npm`: https://github.com/remix-run/remix/pull/8610

Thanks for the PR!  Do you want to make this update in the other templates that use `@typescript-eslint/eslint-plugin` as well?  If not I can do it for the rest once we merge this :)

```
> grep -r "typescript-eslint/eslint-plugin" templates
templates/express/package.json:    "@typescript-eslint/eslint-plugin": "^6.7.4",
templates/fly/package.json:    "@typescript-eslint/eslint-plugin": "^6.7.4",
templates/vite-cloudflare/package.json:    "@typescript-eslint/eslint-plugin": "^6.7.4",
templates/cloudflare-pages/package.json:    "@typescript-eslint/eslint-plugin": "^6.7.4",
templates/spa/package.json:    "@typescript-eslint/eslint-plugin": "^6.7.4",
templates/remix-tutorial/package.json:    "@typescript-eslint/eslint-plugin": "^6.13.0",
templates/vite/package.json:    "@typescript-eslint/eslint-plugin": "^6.7.4",
templates/cloudflare-workers/package.json:    "@typescript-eslint/eslint-plugin": "^6.7.4",
templates/arc/package.json:    "@typescript-eslint/eslint-plugin": "^6.7.4",
templates/vite-express/package.json:    "@typescript-eslint/eslint-plugin": "^6.7.4",
templates/remix/package.json:    "@typescript-eslint/eslint-plugin": "^6.7.4",
```

**Comment by hpnzm** [comment]
Hi @brophdawg11 , 
thanks for the review. I've updated all other templates that use ```@typescript-eslint/eslint-plugin``` like you suggested.

**Review by brophdawg11 — APPROVED** [review]
Thanks!

**Pull request merged**